lib/active_shipping/carriers/usps.rb

```diff
@@ -654,6 +660,71 @@ def parse_tracking_info(response, node)
       )
     end
 
+    def build_address_standardization_request(addresses)
+      xml_builder = Nokogiri::XML::Builder.new do |xml|
+        xml.AddressValidateRequest('USERID' => @options[:login]) do
+        xml.IncludeOptionalElements(true)
+          xml.ReturnCarrierRoute(true)
+          Array(addresses).each_with_index do |address, id|
+            xml.Address('ID' => id) do
+              xml.FirmName(address[:firm_name])
+              xml.Address1(address[:address1])
+              xml.Address2(address[:address2])
+              xml.City(address[:city])
+              xml.State(address[:state])
+              xml.Urbanization(address[:urbanization])
+              xml.Zip5(address[:postal_code])
+              xml.Zip4(address[:zip4])
+            end
+          end
+        end
+      end
+      save_request(xml_builder.to_xml)
+    end
+
+    def parse_address_standardize_response(response, addresses, options = {})
+      success = true
+      message = ''
+      address_hash = {}
+      locations = []
```

Comment:
If you use the `map` assignment suggestion below, you won't need to initialize an empty array variable.